### PR TITLE
Strongly type InterchainGasCalculator

### LINKED ISSUE #407: Strongly type ParsedMessage and InterchainGasCalculator
Atm, `ParsedMessage` is not strongly typed, which prevents InterchainGasCalculator from being strongly typed. We should strongly type ParsedMesage (e.g. use a better type than `string` for origin and destination), and then strongly type InterchainGasCalculator. Be sure to upgrade InterchainGasCalculator docs and tests to use the stronger typing. @yorhodes has some thoughts on work around ParsedMessage, so be sure to chat with him for some context

### PR DESCRIPTION
**Pull request by nambrot**
This PR strongly types InterchainGasCalculator (as well as a dedicated ParsedMessage). This PR also makes InterchainGasCalculator use chain names instead of domain IDs.

I didn't want to override the existing ParsedMessage as that would include having to type AbacusMessage and more and it's not clear to me that we will keep those.

Drive-by changes:
- Removes the sdk's README and points it to the docs
- Adds a changelog file to track breaking changes for the next time sdk gets published

Fixes #407

### CODE REVIEW

File: typescript/sdk/test/gas/calculator.test.ts [hunk 1]
```diff
@@ -206,7 +210,7 @@ describe('InterchainGasCalculator', () => {
         const validatorManager = Object.assign(
           {},
           // @ts-ignore - TODO more strongly type InterchainGasCalculator
-          contracts.inboxes[resolveDomain(originDomain)].validatorManager,
+          contracts.inboxes[resolveDomain(origin)].validatorManager,
```
Comment (tkporter) [inline typescript/sdk/test/gas/calculator.test.ts:213]: I think you can remove the `resolveDomain` here and L224 as well as the ts-ignores

File: typescript/sdk/test/utils.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { FixedNumber, ethers } from 'ethers';
 
-import { NameOrDomain } from '../src';
-import { resolveId } from '../src/core/message';
+import { ChainName } from '../dist';
```
Comment (tkporter) [inline typescript/sdk/test/utils.ts:3]: Should probably be imported from `../src` instead?

**Review by tkporter — APPROVED** [review]
love to see it

**Review by yorhodes — COMMENTED** [review]
amazing, great work!

**Pull request merged**